src/main/java/bio/terra/stairway/Flight.java

```diff
@@ -106,14 +107,14 @@ public void run() {
       logger.debug("Executing: " + context().toString());
       FlightStatus flightStatus = fly();
       flightExit(flightStatus);
-      hookWrapper().endFlight(flightContext);
     } catch (InterruptedException ex) {
       // Shutdown - try disowning the flight
       logger.warn("Flight interrupted: " + context().getFlightId());
       flightExit(READY);
     } catch (Exception ex) {
       logger.error("Flight failed with exception", ex);
     }
+    hookWrapper().endFlight(flightContext);
```

Comment:
unlike `hookWrapper().startFlight`, this isn't wrapped in any `try catch`. Just want to verify that the asymmetry is intended.

Reply:
Good point, I wrapped it. There is nothing to be done besides log on an exception, but probably good policy not to send exceptions out to the thread pool.